core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -564,6 +572,133 @@ protected boolean check(int node, double normedDist, int wayIndex, EdgeIteratorS
         return closestMatch;
     }
 
+    /**
+     * Returns all edges that are within the specified radius around the queried position.
+     *
+     * @param queryLat latitude to search from
+     * @param queryLon longitude to search from
+     * @param edgeFilter only include edges from edgeFilter
+     * @param radius include all results within this radius (in meters) - assuming it's not
+     *        bounded by maxIterations.
+     * @param maxIterations if not null, search only a square of (2 * maxIterations - 1) tiles
```

Comment:
We have to do that even for the normal version but this parameter allows for even longer searches. See https://github.com/graphhopper/graphhopper/pull/221

I would also prefer to avoid this extra parameter in the method - isn't this also more a static configuration like 'maximum allowed visited nodes' and not per request?